### PR TITLE
Support --tag argument

### LINKED ISSUE #2561: Ability to add tag to run to display in 'Runs' list in Dashboard UI
### Current behavior:

Dashboard UI does not show environment

screen shot 2018-10-04 at 9 15 42

### Desired behavior:

Hi,
Is there an option to add environment to the dashboard UI?
So for example if we run sanity tests (or any e2e) for the same project on different environments we have no way to know where did the test ran.

for example screen shot above, test num 100 can run on production, text 101 can run on qa etc...
It would be great to see the env straight on dashboard without entering the test and figuring out where it ran.

### Versions

cypress: v3.1.0
os: CentOS 7.3.1611
browser: Electron 59.0.3071.115

### PR DESCRIPTION
**Pull request by jennifer-shehane**
- close #2561 

### TODO

- [x] Bump json-schema endpoints to 2.2.0 for postRunRequest and postRunResponse
- [x] How should we handle `cypress run --tag nightly` and then running `cypress run --tag staging` in parallel, where the run will become a single run. I think that it should attribute both `staging` and `nightly` to the single run. We would be posting `tags: ["nightly"]` in one POST and then `tags: ["staging"]` in the 2nd POST. https://github.com/cypress-io/cypress-services/issues/1812

### User facing changelog

Cypress now accepts a `--tag` argument to be used to distinguish separate runs. Any tags assigned will be shown on the run in the Cypress Dashboard.

### Additional details

- Accepting `tag` as singular, but always passing it as an Array to the cypress-api as `tags`
- I tested this through staging with the Dashboard and confirmed it does display the tag in the Dashboard, so it is passed along correctly (tested with 1 and multiple tags)

#### Options are alphabetized + Tag option added to `--help`

Screen Shot 2019-11-13 at 1 59 44 PM

#### Error when passing space delimited `--spec`

Screen Shot 2019-11-13 at 1 58 51 PM

#### Error when passing space delimited `--tag`

Screen Shot 2019-11-26 at 3 10 47 PM

#### Tag added to beginning of recording run stdout

Displays as `false` by  default.

1 tag:

Screen Shot 2019-11-26 at 3 04 49 PM

2 tags:

Screen Shot 2019-11-26 at 3 11 49 PM

#### Dashboard displays the tags in the Runs list (only implementation right now)

Screen Shot 2019-11-26 at 3 15 29 PM

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [NA] Have API changes been updated in the [`type definitions`](cli/types/index.d.ts)?
- [NA] Have new configuration options been added to the [`cypress.schema.json`](cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
Need to make an update based on this comment https://github.com/cypress-io/cypress-services/pull/1566#discussion_r339094829 to change sending `tag` to `tags` to the cypress-api

>We are calling it tag to the user in the Test Runner (as this matches the same convention as the spec flag. But it appears when we communicate to the API later, it is changed to specs, so I am fine with us using tags here since I believe we've already normalized it into an array. cypress-io/cypress/pull/5164/files#diff-f558bdd26d5a0b4043bae0b9dc251e9fR291

**Comment by jennifer-shehane** [comment]
I tested this in staging and it is working, so the failed tests are likely just some weird snapshot diffs that need reconciling.

File: packages/server/test/unit/args_spec.coffee [hunk 1]
```diff
@@ -52,6 +52,13 @@ describe "lib/util/args", ->
       options = @setup("--run-project", "foo", "--spec", "'cypress/integration/foo_spec.js'")
       expect(options.spec[0]).to.eq("#{cwd}/cypress/integration/foo_spec.js")
 
+  context "--tag", ->
+    it "converts to array", ->
+      options = @setup("--run-project", "foo", "--tag", "nightly,production,build")
+      expect(options.tag[0]).to.eq("nightly")
```
Comment (bahmutov) [inline packages/server/test/unit/args_spec.coffee]: could we use single assertion here for complete message on error
```js
expect(options.tag).to.deep.equal(['nightly', 'production', 'build'])
```

File: packages/server/test/unit/args_spec.coffee [hunk 2]
```diff
@@ -204,9 +211,9 @@ describe "lib/util/args", ->
 
       ## make sure it works with both --env=foo=bar and --config foo=bar
       @obj = @setup(
+        "--config",
```
Comment (bahmutov) [inline packages/server/test/unit/args_spec.coffee]: ohh, this is the problem - `--config` has moved, but its list of arguments remains in place - causing all the problems!

**Review by bahmutov — APPROVED** [review]
Reviewed, ran tests, added more tests, tested against staging - all seems 👍

File: packages/server/test/unit/args_spec.coffee (hunk 1 above)
Comment (brian-mann) [reply packages/server/test/unit/args_spec.coffee]: yes

File: packages/server/test/unit/args_spec.coffee (hunk 2 above)
Comment (jennifer-shehane) [reply packages/server/test/unit/args_spec.coffee]: Yah, I thought my need for alphabetization may be the culprit, but I couldn’t see it 🤦‍♀️

**Pull request merged**